eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -362,29 +381,41 @@ object MultiPartHandler {
                   }
                 })).map(paths => {
                   val invoiceFeatures = nodeParams.features.bolt12Features()
-                  val invoice = Bolt12Invoice(r.invoiceRequest, paymentPreimage, r.nodeKey, nodeParams.invoiceExpiry, invoiceFeatures, paths.map { case (blindedRoute, paymentInfo, _) => PaymentBlindedRoute(blindedRoute.route, paymentInfo) })
+                  val invoice = Bolt12Invoice(r.invoiceRequest, paymentPreimage, r.nodeKey, nodeParams.invoiceExpiry, invoiceFeatures, paths.map { case (blindedRoute, paymentInfo, _) => PaymentBlindedRoute(blindedRoute.route, paymentInfo) }, r.additionalTlvs, r.customTlvs)
                   log.debug("generated invoice={} for offer={}", invoice.toString, r.invoiceRequest.offer.toString)
-                  nodeParams.db.payments.addIncomingBlindedPayment(invoice, paymentPreimage, paths.map { case (blindedRoute, _, pathId) => blindedRoute.lastBlinding -> pathId.bytes }.toMap, r.paymentType)
+                  if (r.storeInDb) {
+                    nodeParams.db.payments.addIncomingBlindedPayment(invoice, paymentPreimage, paths.map { case (blindedRoute, _, pathId) => blindedRoute.lastBlinding -> pathId }.toMap, r.paymentType)
+                  }
                   invoice
-                }).recover(exception => Status.Failure(exception)).pipeTo(replyTo)
+                }))(WrappedInvoice)
+                forwardInvoice(replyTo)
             }
-            Behaviors.stopped
         }
       }
     }
+
+    def forwardInvoice(replyTo: typed.ActorRef[Try[Invoice]]): Behavior[Command] = {
+      Behaviors.receiveMessagePartial {
+        case WrappedInvoice(invoice) =>
+          replyTo ! invoice
+          Behaviors.stopped
+      }
+    }
   }
 
   object GetIncomingPaymentActor {
 
     // @formatter:off
     sealed trait Command
     case class GetIncomingPayment(replyTo: ActorRef) extends Command
+    case class PaymentFound(payment: IncomingBlindedPayment) extends Command
+    object NoPayment extends Command
```

Comment:
This is rather `RejectPayment`, we also send that command for invalid incoming payment.

Reply:
We try to get an `IncomingPayment` matching the current HTLC and we receive either `PaymentFound` or `NoPayment`. I don't think `RejectPayment` would be a better name here. `RejectPacket` is what this actor returns if it can't find a matching `IncomingPayment`.

Reply:
What I meant is that the `OfferManager` sends this `NoPayment` command in a lot of cases, most of which are for a different reason than the fact there is no matching payment.

Most of the cases where the `OfferManager` sends this command are when the payment we receive is malformed and want to reject it, which indicates that the name is misleading...